compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1220,28 +1229,102 @@ impl<'a> Parser<'a> {
         Ok(impl_info)
     }
 
-    /// Parse `["const" | ("static" "mut"?)] $ident ":" $ty (= $expr)?` with
-    /// `["const" | ("static" "mut"?)]` already parsed and stored in `m`.
+    /// Parse a static or const item with the prefix `"const" | "static" "mut"?` already parsed and stored in `m`.
```

Comment:
That wasn't my idea, it's been like this before and it's hopelessly unreadable. I basically just moved the comment. Let me figure out how this can be improved (splitting `parse_item_global` into two methods would certainly accomplish this but then again, they do share a lot in common esp. if we keep the "generic statics" error recovery logic).